### PR TITLE
[Maulik] use offsets map in commit queue instead of records

### PR DESCRIPTION
**Pull request by mauliksoneji**
Use offsets in commit queue instead of records inorder to reduce the heap.

### CODE REVIEW

File: src/main/java/com/gojek/beast/worker/OffsetCommitWorker.java [hunk 1]
```diff
@@ -13,72 +15,104 @@
...
+                        sleep(defaultSleepMs);
+                    }
+                }
+                statsClient.timeIt("committer.queue.wait.ms", commitQueuePollStartTime);
             }
-            statsClient.timeIt("committer.processing.time", start);
+            commit(partitionsCommitOffset);
+
+            statsClient.gauge("committer.clubbed.offsets", offsetClubbedBatches);
+            statsClient.timeIt("committer.processing.time", startTime);
```
Comment (kushsharma) [inline src/main/java/com/gojek/beast/worker/OffsetCommitWorker.java:102]: Isn't this should be startTime - Instant.now()?

File: src/main/java/com/gojek/beast/worker/OffsetCommitWorker.java [hunk 2]
```diff
@@ -13,72 +15,104 @@
...
+                        log.debug("waiting for {} acknowledgement for offset {}, {}, {}", defaultSleepMs, currentOffset, offsetState.partitionOffsetAckSize(), currentOffset);
+                        sleep(defaultSleepMs);
+                    }
+                }
+                statsClient.timeIt("committer.queue.wait.ms", commitQueuePollStartTime);
             }
-            statsClient.timeIt("committer.processing.time", start);
+            commit(partitionsCommitOffset);
+
+            statsClient.gauge("committer.clubbed.offsets", offsetClubbedBatches);
```
Comment (kushsharma) [inline src/main/java/com/gojek/beast/worker/OffsetCommitWorker.java:101]: What's the thought process behind using a gauge here?

File: src/main/java/com/gojek/beast/sink/OffsetMapQueueSink.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        Instant start = Instant.now();
+        boolean offered;
+        Map<TopicPartition, OffsetAndMetadata> offsetmap = records.getPartitionsCommitOffset();
+        try {
+            offered = recordQueue.offer(offsetmap, config.getTimeout(), config.getTimeoutUnit());
+            statsClient.gauge("queue.elements,name=" + config.getName(), recordQueue.size());
+        } catch (InterruptedException e) {
+            return new FailureStatus(e);
+        }
+        statsClient.gauge("sink.queue.push.messages", offsetmap.size());
```
Comment (kushsharma) [inline src/main/java/com/gojek/beast/sink/OffsetMapQueueSink.java:34]: Why gauge here?

Comment (mauliksoneji) [reply src/main/java/com/gojek/beast/sink/OffsetMapQueueSink.java:34]: these are taken care of in this PR https://github.com/gojek/beast/pull/47

File: src/main/java/com/gojek/beast/worker/OffsetCommitWorker.java (hunk 1 above)
Comment (mauliksoneji) [reply src/main/java/com/gojek/beast/worker/OffsetCommitWorker.java:102]: timeIt function does that only, timeIt function takes in when the timer started and does exactly that.

File: src/main/java/com/gojek/beast/worker/OffsetCommitWorker.java (hunk 2 above)
Comment (mauliksoneji) [reply src/main/java/com/gojek/beast/worker/OffsetCommitWorker.java:101]: This metric will be published after OFFSET_COMMIT_TIME, and the value is the total number of batched offsets, this is not an indication of throughput, its a measure of how many batches were combined. Thus this has to be a gauge metric

File: src/main/java/com/gojek/beast/worker/OffsetCommitWorker.java (hunk 1 above)
Comment (kushsharma) [reply src/main/java/com/gojek/beast/worker/OffsetCommitWorker.java:102]: Got it.

**Comment by kushsharma** [comment]
LGTM

**Pull request merged**